source/ast/types/AllTypes.cpp

```diff
@@ -31,6 +32,55 @@ using namespace slang;
 using namespace slang::syntax;
 using namespace slang::ast;
 
+class InterfaceDefVisitor : public ASTVisitor<InterfaceDefVisitor, true, true> {
+public:
+    InterfaceDefVisitor(const InstanceSymbol& iface, const ASTContext& context) :
+        iface(iface), context(context) {};
+    template<typename T>
+    void handle(const T& symbol) {
+        if constexpr (std::is_base_of_v<Symbol, T>) {
+            if (symbol.kind == SymbolKind::Instance) {
+                const auto* inst = symbol.template as_if<InstanceSymbol>();
```

Comment:
This seems wrong; instantiating an interface inside another interface doesn't make it a port, so we shouldn't error on it.

Reply:
still in progress

Reply:
in the first version of the solution, I misunderstood LRM in this term, sorry

Reply:
add that check in the last commit